### PR TITLE
[Editor] Improve extension palate to provide more information such as syntax and examples 

### PR DESCRIPTION
**Pull request by SinthujanSintha**
>In front end side  I have added some `css` style in order to display extension palate  in proper way,I edited extension data manipulating `js` function with intention to convert `md` data into `html`, changed index.html file with purpose to provide more details in extension palate  ,improved copy to clipboard functions and finally improved the search function of extension operator as well.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

> Earlier extension search bar .
perivous

> Improved extension search bar .
improved

https://github.com/siddhi-io/distribution/pull/731

### CODE REVIEW

File: tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/commons/metadata/ParameterMetaData.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
 
     private String name;
     private List<DataType> type;
+    private Boolean dynamic;
```
[the remark this answers is not in the document]
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/commons/metadata/ParameterMetaData.java]: 7e84bdd611f4506de93c3a41a87dc61bcfa23c6d this commit fix above issue

File: tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/util/SourceEditorUtils.java [hunk 2]
```diff
@@ -434,6 +440,239 @@ private static ProcessorMetaData generateProcessorMetaData(Class<?> processorCla
...
+     *
+     * @param parameterName        parameter name from parameter
+     * @param parameterDataTypeMap Parameter datatype map
+     * @return Parameter Data Type
+     */
+    private static StringBuilder parameterDataTypeGeneration(String parameterName,
+                                                             Map<String, DataType[]> parameterDataTypeMap) {
+
+        StringBuilder parameterDataType = new StringBuilder();
+        parameterDataType.append(" <");
```
[the remark this answers is not in the document]
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/util/SourceEditorUtils.java]: 7e84bdd611f4506de93c3a41a87dc61bcfa23c6d this commit fix above issue

File: tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/util/SourceEditorUtils.java [hunk 3]
```diff
@@ -434,6 +440,239 @@ private static ProcessorMetaData generateProcessorMetaData(Class<?> processorCla
...
+     * generate the syntax  for source and sink extension.
+     *
+     * @param extension            Extension data from extension anotation
+     * @param parameterDataTypeMap Parameter datatype map
+     * @return syntaxList
+     */
+    private static List<SyntaxMetaData> sourceSinkSyntaxGeneration(Extension extension,
+                                                                   Map<String, DataType[]> parameterDataTypeMap
+                                                                  ) {
+
```
[the remark this answers is not in the document]
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/util/SourceEditorUtils.java]: 7e84bdd611f4506de93c3a41a87dc61bcfa23c6d this commit fix above issue

**Comment by mohanvive** [comment]
@pcnfernando Are we good to merge this?

**Pull request merged**